src/tests/background.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from 'chai'
+import { createMocks, Mocks } from './mocks'
+import { run } from '../background/run'
+import { ensureActiveTab } from '../selectors'
+
+describe('background:run', () => {
+  let mocks: Mocks // tslint:disable-line:no-let
+
+  before(() => {
+    mocks = createMocks()
+    run(mocks.window, mocks.browser, mocks.chrome)
+  })
+
+  describe('happy path', () => {
+    it('loads window.store, which starts with an empty state', () => {
+      expect(mocks.window).to.have.property('store')
+      expect(mocks.window.store.getState()).to.eql({
+        popupConnected: false,
+        focusedWindowId: -1,
+        tabs: new Map(),
+        toBeTweeted: null,
+        justTweeted: null,
+        twitterAuth: 'not_authed',
+        pickingEmoji: false,
+        alert: null,
+        mostRecentAction: { type: 'INITIALIZING' },
+      })
+    })
+
+    it('sets the focusedWindowId when chrome.windows.getAll calls back', () => {
+      expect(mocks.chrome.windows.getAll).to.have.callCount(1)
+      const [callback] = mocks.chrome.windows.getAll.firstCall.args
+      callback([
+        { id: 1, focused: false },
+        { id: 2, focused: true },
+        { id: 3, focused: false },
+      ])
+      expect(mocks.window.store.getState()).to.have.property('focusedWindowId', 2)
+    })
+
+    it('sets the tabs when chrome.tabs.query calls back', () => {
```

Comment:
We're expecting the previous test to run before this test, but I don't know if it's a good idea to build our test suite this way. If we do, we should at least document it.